### PR TITLE
Add dockerfile parameter

### PR DESCRIPTION
**Pull request by TrevCraw**

### CODE REVIEW

**Comment by TrevCraw** [comment]
I ran into quite a few local test errors, however, I am not sure how these changes could have caused the errors.
```
C:\Users\TREVORCrawford\Git\ci.gradle>gradle build -Druntime=ol -DruntimeVersion=20.0.0.1

> Task :test

io.openliberty.tools.gradle.DevTest > classMethod FAILED
    java.lang.AssertionError at DevTest.groovy:302

io.openliberty.tools.gradle.InstallDirSubProject > testProjectDirectoriesPostClean FAILED
    org.codehaus.groovy.runtime.powerassert.PowerAssertionError at InstallDirSubProject.groovy:53

io.openliberty.tools.gradle.PollingDevTest > classMethod FAILED
    java.lang.AssertionError at PollingDevTest.groovy:43

io.openliberty.tools.gradle.PollingDevTest > classMethod FAILED
    java.io.IOException

io.openliberty.tools.gradle.TestLooseEarApplication > test_start_with_timeout_success FAILED
    java.lang.AssertionError at TestLooseEarApplication.groovy:116

io.openliberty.tools.gradle.TestSpringBootApplication15 > test_spring_boot_dropins_15 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication15.groovy:78

io.openliberty.tools.gradle.TestSpringBootApplication15 > test_spring_boot_classifier_apps_15 FAILED
    java.lang.AssertionError at TestSpringBootApplication15.groovy:69
    org.gradle.testkit.runner.UnexpectedBuildFailure at TestSpringBootApplication15.groovy:40

io.openliberty.tools.gradle.TestSpringBootApplication15 > test_spring_boot_apps_15 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication15.groovy:48

io.openliberty.tools.gradle.TestSpringBootApplication20 > test_spring_boot_war_apps_20 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication20.groovy:82

io.openliberty.tools.gradle.TestSpringBootApplication20 > test_spring_boot_dropins_20 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication20.groovy:113

io.openliberty.tools.gradle.TestSpringBootApplication20 > test_spring_boot_war_classifier_apps_20 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication20.groovy:98

io.openliberty.tools.gradle.TestSpringBootApplication20 > test_spring_boot_classifier_apps_20 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication20.groovy:66

io.openliberty.tools.gradle.TestSpringBootApplication20 > test_spring_boot_apps_20 FAILED
    org.junit.ComparisonFailure at TestSpringBootApplication20.groovy:50

112 tests completed, 13 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///C:/Users/TREVORCrawford/Git/ci.gradle/build/reports/tests/test/index.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1h 0m 27s
8 actionable tasks: 5 executed, 3 up-to-date
```

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy [hunk 1]
```diff
@@ -166,6 +166,14 @@ class DevTask extends AbstractServerTask {
         this.container = container;
     }
 
+    private File dockerfile;
+
+    @Option(option = 'dockerfile', description = 'Dev mode will build a docker image from the provided Dockerfile and start a container from the new image.')
+    void setDockerfile(String dockerfile) {
+        File tmp = new File(dockerfile);
+        this.dockerfile = new File(tmp.getCanonicalPath()); // ensures the dockerfile is defined with the full path - matches how maven behaves
```
Comment (ericglau) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy:174]: `getCanonicalPath()` could throw IOException, consider whether we should catch that and rethrow an exception with a more user-oriented message.

Comment (TrevCraw) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy]: Updated

File: src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy [hunk 2]
```diff
@@ -207,13 +207,13 @@ class DevTask extends AbstractServerTask {
...
-                    int verifyAppStartTimeout, int appUpdateTimeout, double compileWait, 
+                    int verifyAppStartTimeout, int appUpdateTimeout, double compileWait,
                     boolean libertyDebug, boolean pollingTest, boolean container, File dockerfile
         ) throws IOException {
             super(serverDirectory, sourceDirectory, testSourceDirectory, configDirectory, projectDirectory,
                     resourceDirs, hotTests, skipTests, false, false, artifactId,  serverStartTimeout,
-                    verifyAppStartTimeout, appUpdateTimeout, ((long) (compileWait * 1000L)), libertyDebug, 
-                    true, true, pollingTest, container, dockerfile);
+                    verifyAppStartTimeout, appUpdateTimeout, ((long) (compileWait * 1000L)), libertyDebug,
+                    true, true, pollingTest.booleanValue(), container.booleanValue(), dockerfile);
```
Comment (ericglau) [inline src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy:216]: Why does this need .booleanValue()?  pollingTest and container are primitive booleans here

Comment (turkeylurkey) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy:216]: I made this mistake too. I thought we wanted the class variable in all cases but in this case we need to refer to the parameter of the DevTaskUtil constructor.

Comment (TrevCraw) [reply src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy:216]: The `.booleanValue()` additions were changes pulled into my branch. Do you want them removed?

**Pull request merged**